modules/sfp_stackoverflow.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_stackoverflow
+# Purpose:      Example module to use for new modules.
+#
+# Author:      Jess Williams <jesscia_williams0@protonmail.com>
+#
+# Created:     2021-12-12
+# Copyright:   (c) Steve Micallef
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import re
+import netaddr
+
+
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+class sfp_stackoverflow(SpiderFootPlugin):
+
+    meta = {
+        'name': "Stackoverflow",
+        'summary': "Search StackOverflow for any mentions of a target domain. Returns potentially related information.",
+        'flags': ["errorprone"],
+        'useCases': ["Passive"],
+        'categories': ["Content Analysis"],
+        'dataSource': {
+            'website': "https://www.stackecxchange.com",
+            'model': "FREE_NOAUTH_LIMITED",
+            'references': [
+                "https://api.stackexchange.com/docs"
+            ],
+            'apiKeyInstructions': [
+                "Visit https://stackapps.com/apps/oauth/register",
+            ],
+            'favIcon': "https://cdn.sstatic.net/Sites/stackoverflow/Img/favicon.ico?v=ec617d715196",
+            'logo': "https://cdn.sstatic.net/Sites/stackoverflow/Img/apple-touch-icon.png",
+            'description': "A paragraph of text with details about the data source / services. "
+            "Keep things neat by breaking the text up across multiple lines as "
+            "has been done here. If line breaks are needed for breaking up "
+            "multiple paragraphs, use \n.",
+        }
+    }
+
+    # Default Options
+    opts = {
+        'api_key': '',
+    }
+
+    # Option descriptions
+    optdescs = {
+        "api_key": "StackApps Optional API Key."
```

Comment:
What limits are imposed if not provided? Would be good to have a few words about that here (nit: Optional should be all lower case).